executor.go

```diff
@@ -566,12 +566,11 @@ func (e *executor) executeBitmapCallShard(ctx context.Context, index string, c *
 }
 
 // executeSumCountShard calculates the sum and count for bsiGroups on a shard.
-func (e *executor) executeSumCountShard(ctx context.Context, index string, c *pql.Call, shard uint64) (ValCount, error) {
+func (e *executor) executeSumCountShard(ctx context.Context, index string, c *pql.Call, filter *Row, shard uint64) (ValCount, error) {
 	span, ctx := tracing.StartSpanFromContext(ctx, "Executor.executeSumCountShard")
 	defer span.Finish()
 
-	var filter *Row
-	if len(c.Children) == 1 {
+	if filter == nil && len(c.Children) == 1 {
```

Comment:
a comment explaining the difference between the children and the explicit filter arg would not go awry.